### ISSUE TITLE
Kali linux socket: Permission denied, run ettercap failed

### ISSUE THREAD

**Issue by Neutrino3316** [CypherpunkArmory/UserLAnd#438]
Device: [e.g. Pixel XL 2]
Android Version: 8.0.0
UserLAnd Version 1.0.2

Run UserLAnd
Click on `Apps`
Click on `Kali`
Then wait till it is installed
Next, I connect to this kali by ssh

## ettercap can't run

I install ettercap by this:

```bash
sudo apt-get install zlib1g zlib1g-dev
sudo apt-get install build-essential
sudo apt-get install ettercap-text-only
```

add try to run ettercap

```bash
ettercap -T
```

and then get this error:

```
ettercap 0.8.2 copyright 2001-2015 Ettercap Development Team

ERROR : 13, Permission denied
[/build/ettercap-sGGsY1/ettercap-0.8.2/src/ec_capture.c:capture_getifs:114]

 socket: Permission denied
```

![image](https://user-images.githubusercontent.com/20875523/49330205-703a5180-f5c6-11e8-87df-163caedf084e.png)

## netdiscover can't run

I install netdiscover by this:

```bash
sudo apt-get install netdiscover
```

add try to run netdiscover

```bash
netdiscover -i wlan0
```

and then get this error:

```
pcap_open_live(): socket for SIOCETHTOOL(ETHTOOL_GET_TS_INFO): Permission denied
```

![image](https://user-images.githubusercontent.com/20875523/49330390-e2ac3100-f5c8-11e8-838a-b63ba1ec5566.png)

Am I missing something? Or does anyone have a good idea to solve this problem?

**Comment by EnderNightLord-ChromeBook** [comment]
you may need to use:

$ sudo su (to go into root mode) and then use your command
or
$ sudo (and your command)

**Comment by Neutrino3316** [comment]
> you may need to use:
> 
> $ sudo su (to go into root mode) and then use your command
> or
> $ sudo (and your command)

Still, it can't work. I get the error `Permission denied` again.
Maybe that's because my phone is no rooted?

**Comment by Neutrino3316** [comment]
> no... the file system has access to itself so I don't know...

I think maybe it's because that we don't have high privilege authorization to access and control the network device.

**Comment by EnderNightLord-ChromeBook** [comment]
you might be right there...

**Comment by EwanValentine** [comment]
Anyone find a solution for this?

**Comment by EnderNightLord-ChromeBook** [comment]
Um not yet... It may be the program just can't run in a chroot.

**Comment by EwanValentine** [comment]
I've read it's possibly missing ath10k drivers or something? I'm going to try a usb wifi dongle and see what happens

**Comment by jawetzel1992** [comment]
Same issue with my Samsung s9 plus, I'm thinking need external wifi card that's compatible. Anyone else try one yet with same issues ?

**Comment by antuspcm** [comment]
same on my s9 with ping. strace showed permission denied opening a socket with admin and raw capabilities. found a workaround in another thread so repost here for orthers. run su (even though you are allready root) and the new shell is initialised properly and then it works. looks like a kali bug, nothing to do with your hardware.

**Comment by T16bros** [comment]
i did sudo su
then sudo ettercad -G and it worked for me

**Comment by k1941996** [comment]
i dont think this is a permanent solution. Im running arp-scan --local then also erro arrives: my_lookupdev: Permission Denied

**Comment by PrincessNikki** [comment]
Same issue on my Samsung Galaxy Note 9

**Comment by L96Github** [comment]
> i dont think this is a permanent solution. Im running arp-scan --local then also erro arrives: my_lookupdev: Permission Denied

Yep same issue on Huawei P30 Pro. `su sudo` didn't work in either arp-scan nor netdiscover

**Comment by YeMinAung2004** [comment]
i'm stucking on this i'm using Android kali linux but ettercap says this : ERROR : 13, Permission denied
[/build/ettercap-gsX8lN/ettercap-0.8.3.1/src/ec_capture.c:capture_getifs:114]

 ccmni0: SIOCETHTOOL(ETHTOOL_GLINK) ioctl failed: Permission denied

**Comment by Deepanshu99-tech** [comment]
Any one have any kind of solution for regarding that problem.
So please suggest me. Because of many time due to this my hole work stopped.

**Comment by Rozer-12** [comment]
@Deepanshu99-tech Have you got any answer please suggest I am facing the same

**Comment by Babunu** [comment]
pcap_open_live(): socket for SIOCETHTOOL(ETHTOOL_GET_TS_INFO): Permission denied


Please solve this error

**Comment by emirhanengineer** [comment]
I got it.

**Comment by corbinlc** [comment]
Closing as part of a repository cleanup. If this issue is still relevant, feel free to reopen it or file a fresh one.
